src/evo/deterministicmns.cpp

```diff
@@ -1126,7 +1126,8 @@ bool CDeterministicMNManager::UpgradeDBIfNeeded()
     LOCK(cs_main);
 
     if (chainActive.Tip() == nullptr) {
-        return true;
+        // should have no records
+        return !evoDb.GetRawDB().Exists(std::string("b_b")) && !evoDb.GetRawDB().Exists(EVODB_BEST_BLOCK);
```

Comment:
```suggestion
        return evoDb.IsEmpty();
```

I guess we can just use `IsEmpty` here too then?